addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -657,6 +657,87 @@ public void OnEntityCreated(int iEntity, const char[] sClassname)
 	
 	if (StrEqual(sClassname, "tf_dropped_weapon") || StrEqual(sClassname, "item_powerup_rune"))	//Never allow dropped weapon and rune dropped from survivors
 		RemoveEntity(iEntity);
+		
+	if (StrEqual(sClassname, "prop_dynamic") && g_nRoundState == SZFRoundState_Active)
+	{
+		RequestFrame(Frame_SetSpawnedWeapon, iEntity);
```

Comment:
Can go for `SDKHook_Spawn` or `SDKHook_SpawnPost` rather than needing to wait a frame for weapon to be valid,